test/tools/fuzzing/fuzz.py

```diff
@@ -0,0 +1,129 @@
+            if file_matcher(absolute_path):
+                return absolute_path
+        else:
+            # path does not exist
+            continue
+    raise ValueError("Couldn't find {}".format(debug_file_name))
+
+def find_or_use_user_supplied_path(test_base_ref_dir, user_supplied_bin_path, search_paths, is_correct_bin_func, debug_file_name):
+    if user_supplied_bin_path is None:
+        return find_in_path(search_paths, is_correct_bin_func, debug_file_name)
+    else:
+        if not Path(user_supplied_bin_path).exists() or not Path(user_supplied_bin_path).is_file():
+            raise ValueError("Invalid {debug_file_name} binary path: {}".format(
+                (user_supplied_bin_path)))
+        return user_supplied_bin_path
+    
+def find_test_binary(test_base_ref_dir, user_supplied_bin_path):
+    vw_search_paths = [
+        Path(test_base_ref_dir).joinpath("../../../build/test/tools/fuzzing")
+    ]
+
+    def is_test_binary(file_path):
+        file_name = os.path.basename(file_path)
+        return file_name == "test_initialize"
+
+    return find_or_use_user_supplied_path(
+        test_base_ref_dir=test_base_ref_dir,
+        user_supplied_bin_path=user_supplied_bin_path,
+        search_paths=vw_search_paths,
+        is_correct_bin_func=is_test_binary,
+        debug_file_name="test_initialize")
+
+def main():
+    
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('--afl_dir', required=True, help="Directory of where the built outputs for google AFL are located")
+    parser.add_argument(
+        '--test_bin_path', help="Specify test binary to use. Otherwise, binary will be searched for in build directory")
+    parser.add_argument(
+        '-t', '--timeout', type=int, default=60,  help="Max runtime for each fuzzing test in minutes. 0 will disable the timeout")
+    parser.add_argument(
+        '-f', '--full_tests',  action='store_true', help="Run full tests. This disables the '-d' option in AFL")
+    #parser.add_argument(
+    #    '-p', '--parallel', type=int, default=1,  help="Number of CPUs to use per test.")
+    args = parser.parse_args()
+
+    # Unfortunately, there doesn't appare to be a sane way to run different tests concurrently. Therefore this script will
+    # run for (# of tests * timeout) seconds
+    
+    script_dir = Path(os.path.dirname(os.path.abspath(__file__)))
+    test_bin = find_test_binary(script_dir, args.test_bin_path)
+    # AFL requires a separate read-only dir for the models. I believe it defaults to stdin inputs unless you define
+    # the '@@' parameter in the command line
+    models_dir = os.path.join(script_dir, 'models')
+    
+    print("Using VW binary: {}".format((test_bin)))
+
+    try:
+        os.mkdir('output')
+    except FileExistsError:
+        pass
+
+    # Create a temporary working directory for any output files VW produces
+    try:
+        os.mkdir('tmpwd')
```

Comment:
You can a temp dir to avoid having to clear it between runs. https://stackoverflow.com/questions/3223604/how-to-create-a-temporary-directory-and-get-its-path-file-name

Reply:
This doesn't clear anything out, it just creates a temp working directory and runs the tests in there. I suppose I could use tempfile, but it might be nice to keep it around to see what cache files get generated.